pallets/gear-program/src/migrations/allocations.rs

```diff
@@ -0,0 +1,293 @@
+        traits::Saturating,
+    };
+    use sp_std::{collections::btree_set::BTreeSet, prelude::*};
+
+    #[derive(Clone, Debug, Decode, Encode, PartialEq, Eq, TypeInfo)]
+    #[codec(crate = codec)]
+    #[scale_info(crate = scale_info)]
+    pub struct ActiveProgram<BlockNumber: Copy + Saturating> {
+        pub allocations: IntervalsTree<WasmPage>,
+        pub pages_with_data: IntervalsTree<GearPage>,
+        pub memory_infix: MemoryInfix,
+        pub gas_reservation_map: GasReservationMap,
+        pub code_hash: H256,
+        pub code_exports: BTreeSet<DispatchKind>,
+        pub static_pages: WasmPage,
+        pub state: ProgramState,
+        pub expiration_block: BlockNumber,
+    }
+
+    #[derive(Clone, Debug, Decode, Encode, PartialEq, Eq, TypeInfo)]
+    #[codec(crate = codec)]
+    #[scale_info(crate = scale_info)]
+    pub enum Program<BlockNumber: Copy + Saturating> {
+        Active(ActiveProgram<BlockNumber>),
+        Exited(ProgramId),
+        Terminated(ProgramId),
+    }
+
+    #[cfg(feature = "try-runtime")]
+    use {
+        crate::{Config, Pallet},
+        frame_support::{
+            storage::types::StorageMap,
+            traits::{PalletInfo, StorageInstance},
+            Identity,
+        },
+        sp_std::marker::PhantomData,
+    };
+
+    #[cfg(feature = "try-runtime")]
+    pub struct ProgramStoragePrefix<T>(PhantomData<T>);
+
+    #[cfg(feature = "try-runtime")]
+    impl<T: Config> StorageInstance for ProgramStoragePrefix<T> {
+        const STORAGE_PREFIX: &'static str = "ProgramStorage";
+
+        fn pallet_prefix() -> &'static str {
+            <<T as frame_system::Config>::PalletInfo as PalletInfo>::name::<Pallet<T>>()
+                .expect("No name found for the pallet in the runtime!")
+        }
+    }
+
+    #[cfg(feature = "try-runtime")]
+    pub type ProgramStorage<T> = StorageMap<
+        ProgramStoragePrefix<T>,
+        Identity,
+        ProgramId,
+        Program<frame_system::pallet_prelude::BlockNumberFor<T>>,
+    >;
+}
+
+#[cfg(test)]
+#[cfg(feature = "try-runtime")]
+mod test {
+    use super::*;
+    use crate::mock::*;
+    use common::GearPage;
+    use frame_support::traits::StorageVersion;
+    use frame_system::pallet_prelude::BlockNumberFor;
+    use gear_core::{ids::ProgramId, pages::WasmPage, program::ProgramState};
+    use sp_runtime::traits::Zero;
+
+    #[test]
+    fn migration_works() {
+        env_logger::init();
+
+        new_test_ext().execute_with(|| {
+            StorageVersion::new(MIGRATE_FROM_VERSION).put::<GearProgram>();
+
+            // add active program
+            let active_program_id = ProgramId::from(1u64);
+            let program = v5::Program::<BlockNumberFor<Test>>::Active(v5::ActiveProgram {
+                allocations: [1u16, 2, 3, 4, 5, 101, 102]
+                    .into_iter()
+                    .map(WasmPage::from)
+                    .collect(),
+                pages_with_data: [4u16, 5, 6, 7, 8, 400, 401]
```

Comment:
arent' pages with data were always in allocations?

Reply:
Yes they are. But this is GearPages that have nnumbers 4 times bigger